app/models/user.rb

```diff
@@ -66,6 +66,10 @@ def metrics(days)
     return array
   end
 
+  def self.with_role(role_name)
```

Comment:
I did used to write all scopes like that, but not too long ago re-read the guides about scopes with arguments: http://guides.rubyonrails.org/active_record_querying.html#passing-in-arguments and now will write scopes with arguments as class methods.  Either is fine to my eyes, but if one method is 'preferred' by the community, I'd stick with that just to keep code consistent.

Joins shouldn't be shied away from, but you're right they can cause problems if you're not a little bit cautious. In this case we're joining onto a table of ~7 rows I think, so there really shouldn't be an issue. Plus the roles are unlikely to expand into anything that would cause a performance issue. Using `joins` would be less of a performance issue than `includes` in this case, which is another pattern I see from time to time, another alternative would be using a subquery, but then you're starting to get into convoluted code and premature optimisation, besides which I doubt it would be any more performant anyway. :)

Tests, my general methodology is to use integration (feature, system) tests, and unit test any important or complicated logic models (and other lower level classes such as services). I generally don't bother with such simple scopes as this. I am however more than open to a discussion on testing methodology for this project - I am also new to the codebase, and bringing it under a solid test suite is one of the things that I'm looking forward to. In fact if we're both going to be working on this a discussion of testing methodology before any serious work starts would be very beneficial!

Thanks for the comments. :)